volume-providers/vpc/provider/provider.go

```diff
@@ -77,6 +77,8 @@ func NewProvider(conf *config.Config, logger *zap.Logger) (local.Provider, error
 		return nil, errors.New("Incomplete config for VPCBlockProvider")
 	}
 
+	authConfig := &config.BluemixConfig{}
+
 	//Do config validation and enable only one generationType (i.e VPC-Classic | VPC-NG)
 	gcConfigFound := (conf.VPC.EndpointURL != "") && (conf.VPC.TokenExchangeURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
 	g2ConfigFound := (conf.VPC.G2EndpointURL != "") && (conf.VPC.G2TokenExchangeURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
```

Comment:
```	
conf.VPC.EndpointURL = conf.VPC.G2EndpointURL
conf.VPC.TokenExchangeURL = conf.VPC.G2TokenExchangeURL
```

Are you not overriding the privateEndpoint here  for NG ?  OR NG will have either private OR public endpoint populated  in secrets